### DISCUSSION TITLE
decwar.tap install is go:)

### DISCUSSION THREAD

**Discussion by statespacedev** [drforbin/decwar#21] [General]
lots of progress today!:)
figured out tons around tap files and tape drives - working!
will write up some notes by tomorrow 

time to start looking at getting all latest fixes etc into the source files:)

**Comment by statespacedev** [comment]
alright last bizarre thing about decwar.tap install bypassed - it's workn full on now - 
we've got complete end to end scripted 'decwar machine' from tape!:)
just gonna fly the freak flag here on what cost several hours today -
this could be a simh thing?

in this here

    expect "\r\n." send "compile @com1\r"; continue
    expect "\r\n." send "compile @com2\r"; continue
    expect "\r\n." send "r link\r"; continue
    expect "\r\n*" send "@can1\r"; continue
    expect "\r\n." send ""; continue
    expect "\r\n." send "get decwar\r"; continue
    expect "\r\n." send "ssave\r"; continue
    expect "\r\n." send "k/f\r"; continue

if you don't do the send "" line you get a bizarre 'race condition' where the next line beats the '@can1\r' send to the linker - 
iterated on this many tens of times, it's real and it's weird - tried 'pauses', 'delays', 'sleeps', etc
only real workaround was to 'win the race' with 'nothing'!:0

**Comment by drforbin** [comment]
using expect huh? very cool

**Reply by statespacedev** [reply]
learned it all copying from rich's scripts! didn't see anything like the bizarre need to 'send nothing' though - could be 'all mine' hahaha!:)

**Comment by drforbin** [comment]
FYI. I think I used back10 (there are others) I'll look them. 
Try ./backwr -c -f ./mud86/MUD1/* -T >mud.tap
I think I used that to create the tap file I submitted to ut brisco archive. The tops-10 version

**Reply by statespacedev** [reply]
back10 rocked - no probs with it at all - worked out of the box - 99% of the pain today was that dang 'race condition' for the linker - just so weird, seeing this kinda 'async' behaviour!:(

**Comment by statespacedev** [comment]
working scripts committed here - will still be cleaning em up over next few days -
https://github.com/drforbin/decwar/tree/minimalist/docs/tops703
onward to 'latest greatest code' going into the tap file:)

**Comment by statespacedev** [comment]
fwiw fyi - new 'summary notes page' here!:)
https://github.com/drforbin/decwar/blob/minimalist/docs/sec7-decwar-tape-file.md
